### ISSUE TITLE
Regex runs infinitely

### ISSUE THREAD

**Issue by Famlam** [jquery/sizzle#87]
EDIT: apparently github can't handle inline HTML. The correct post is also placed here:
http://pastebin.com/sbJ5h9R5
======================old, broken message===================
Try the following code in Google Chrome:
<html><head><script src="sizzle.js"></script><script>
Sizzle("[onclick^*=\"window.open('http://www.firstload.de/\"]");
alert('Done'); // Never fired
</script></head><body></body></html>
The 'Done' alert will never be fired. This is because m = chunker.exec( soFar ); never finishes.
The cause of this issue, and a fix is explained in http://code.google.com/p/v8/issues/detail?id=1888 by a Chrome developer.

Basically it's just a matter of getting rid of the (3) instances of "+)+" in the regex (which are all non-capturing, so it doesn't change the results of exec either):

chunker, currently
/((?:((?:([^()]+)|[^()]+)+)|[(?:[[^[]]_]|['"][^'"]_['"]|[^[]'"]+)+]|\.|[^ >+~,([\]+)+|[>+~])(\s_,\s_)?((?:.|\r|\n)_)/g
would become
/((?:((?:([^()]+)|[^()])+)|[(?:[[^[]]_]|['"][^'"]_['"]|[^[]'"])+]|\.|[^ >+~,([\])+|[>+~])(\s_,\s_)?((?:.|\r|\n)_)/g
